src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.MultiSegment.cs

```
@@ -0,0 +1,2408 @@
+
+                SequencePosition startPosition = _currentPosition;
+                int startConsumed = _consumed;
+                int alreadyMatched = literal.Length - leftToMatch.Length;
+                while (true)
+                {
+                    _totalConsumed += alreadyMatched;
+                    _bytePositionInLine += alreadyMatched;
+                    if (!GetNextSpan())
+                    {
+                        _totalConsumed = prevTotalConsumed;
+                        consumed = default;
+                        if (IsLastSpan)
+                        {
+                            goto Throw;
+                        }
+                        return false;
+                    }
+
+                    span = _buffer;
+
+                    if (span.StartsWith(leftToMatch))
+                    {
+                        HasValueSequence = true;
+                        SequencePosition start = new SequencePosition(startPosition.GetObject(), startPosition.GetInteger() + startConsumed);
+                        SequencePosition end = new SequencePosition(_currentPosition.GetObject(), _currentPosition.GetInteger() + leftToMatch.Length);
+                        ValueSequence = _sequence.Slice(start, end);
+                        consumed = leftToMatch.Length;
+                        return true;
+                    }
+
+                    if (!leftToMatch.StartsWith(span))
+                    {
+                        _bytePositionInLine += FindMismatch(span, leftToMatch);
+                        goto Throw;
+                    }
+
+                    leftToMatch = leftToMatch.Slice(span.Length);
+                    alreadyMatched = span.Length;
+                }
+            }
+        Throw:
+            _totalConsumed = prevTotalConsumed;
+            consumed = default;
+            ThrowInvalidLiteralMultiSegment(span);
+            return false;
```

This is the only call site for ThrowInvalidLiteralMultiSegment.  Why not have it return the exception to throw and then use `throw` here rather than having the unreachable `return false`?